src/nfs/metadata/dir_metadata.rs

```diff
@@ -0,0 +1,172 @@
+            try!(e.emit_struct_field("name", 2, |e| self.name.encode(e)));
+            try!(e.emit_struct_field("created_time_sec", 3, |e| created_time.sec.encode(e)));
+            try!(e.emit_struct_field("created_time_nsec", 4, |e| created_time.nsec.encode(e)));
+            try!(e.emit_struct_field("modified_time_sec", 5, |e| modified_time.sec.encode(e)));
+            try!(e.emit_struct_field("modified_time_nsec", 6, |e| modified_time.nsec.encode(e)));
+            try!(e.emit_struct_field("user_metadata", 7, |e| self.user_metadata.encode(e)));
+
+            Ok(())
+        })
+    }
+}
+
+impl Decodable for DirMetadata {
+    fn decode<D: Decoder>(d: &mut D) -> Result<DirMetadata, D::Error> {
+        d.read_struct("DirMetadata", 8, |d| {
+            Ok(DirMetadata {
+                locator: try!(d.read_struct_field("locator", 0, Decodable::decode)),
+                encrypt_key: try!(d.read_struct_field("encrypt_key", 1, Decodable::decode)),
+                name: try!(d.read_struct_field("name", 2, Decodable::decode)),
+                created: ::time::at_utc(::time::Timespec {
+                    sec: try!(d.read_struct_field("created_time_sec", 3, Decodable::decode)),
+                    nsec: try!(d.read_struct_field("created_time_nsec", 4, Decodable::decode)),
+                }),
+                modified: ::time::at_utc(::time::Timespec {
+                    sec: try!(d.read_struct_field("modified_time_sec", 5, Decodable::decode)),
+                    nsec: try!(d.read_struct_field("modified_time_nsec", 6, Decodable::decode)),
+                }),
+                user_metadata: try!(d.read_struct_field("user_metadata", 7, Decodable::decode)),
+            })
+        })
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use maidsafe_utilities::serialisation::{deserialise, serialise};
+    use rand;
+    use super::*;
+
+    #[test]
+    fn serialise_directory_metadata_without_parent_directory() {
+        let id = rand::random();
+        let obj_before = DirMetadata::new(id, "hello.txt".to_string(), Vec::new(), None);
+        let serialised_data = unwrap!(serialise(&obj_before));
+        let obj_after = unwrap!(deserialise(&serialised_data));
+        assert_eq!(obj_before, obj_after);
+    }
+
+    #[test]
+    fn update_using_setters() {
+        let modified_time = ::time::now_utc();
+        let id = rand::random();
+        let mut obj_before = DirMetadata::new(id, "hello.txt".to_string(), Vec::new(), None);
+        let user_metadata = "{mime: \"application/json\"}".to_string().into_bytes();
+        obj_before.set_user_metadata(user_metadata.clone());
+        obj_before.set_modified_time(modified_time);
+        obj_before.set_name("index.txt".to_string());
+        let serialised_data = unwrap!(serialise(&obj_before));
+        let obj_after: DirMetadata = unwrap!(deserialise(&serialised_data));
+        assert_eq!(*user_metadata, *obj_after.user_metadata());
```

Comment:
Since it implements `Eq and PartialEq` maybe simply use `assert_eq!(dir_meta_before, dir_meta_after)` instead of for each field